internal/services/containers/kubernetes_cluster_data_source.go

```diff
@@ -380,6 +380,31 @@ func dataSourceKubernetesCluster() *pluginsdk.Resource {
 
 			"identity": commonschema.SystemOrUserAssignedIdentityComputed(),
 
+			"key_vault_kms": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"enabled": {
+							Type:     pluginsdk.TypeBool,
+							Computed: true,
+						},
+						"key_id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"key_vault_network_access": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"key_vault_resource_id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
```

Comment:
I would recommend adding `ValidateFunc` like [`azurerm_key_vault_key](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/keyvault/key_vault_key_resource.go#L59-L64) did.

Maybe `key_vault_id` would be a better name for this argument?